neptune/management/internal/utils.py

```diff
@@ -13,3 +13,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import re
+from typing import Optional
+
+from neptune.management.exceptions import MissingWorkspaceName, ConflictingWorkspaceName
+from neptune.patterns import PROJECT_QUALIFIED_NAME_PATTERN
+
+
+def normalize_project_name(name: str, workspace: Optional[str] = None):
+    project_spec = re.search(PROJECT_QUALIFIED_NAME_PATTERN, name)
```

Comment:
this can return `None` if `name` doesn't match `PROJECT_QUALIFIED_NAME_PATTERN` at all (e.g. `"this/project/is/wrong"`)

please handle that graciously (`ManagementOperationFailure`) and add a unit test :)